### PR TITLE
feat: add recipe details component

### PR DESCRIPTION
**Pull request by StevieShibly8**
- gh-31: Add /api/ route to api calls
- Add recipe details component
- Add recipe details controller that gets recipe data from the database
- Add Recipe and Ingredient data classes
- Add DbSet for Recipe and Ingredient classes to ApplicationDbContext

#19

### CODE REVIEW

**Review by NilinReza — APPROVED** [review]
Hope it works! :)

File: client/src/app/app.module.ts [hunk 1]
```diff
@@ -37,7 +39,8 @@ import { CommonModule, NgFor, NgIf, UpperCasePipe } from '@angular/common';
       { path: '', component: HomeComponent, pathMatch: 'full' },
       { path: 'counter', component: CounterComponent },
       { path: 'fridge', component: FridgeComponent },
-      { path: 'fetch-data', component: FetchDataComponent, canActivate: [AuthorizeGuard] },
+      { path: 'fetch-data', component: FetchDataComponent },
+      { path: 'recipe-details', component: RecipeDetailsComponent },
```
Comment (zaidi-01) [inline client/src/app/app.module.ts:43]: Just `recipe` fits better, `recipe/{recipe_id}`.

File: client/src/app/recipe-details/recipe-details.component.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+export class RecipeDetailsComponent {
+
+    public recipeDetails: RecipeDetails;
+
+    
+// Need to change this so recipe details are fetched only after selecting specific recipe from recipe search results
+    constructor(http: HttpClient, @Inject('BASE_URL') baseUrl: string) {
+        this.recipeDetails = {} as RecipeDetails;
+        http.get<RecipeDetails>(baseUrl + 'api/recipedetails').subscribe(result => {
```
Comment (zaidi-01) [inline client/src/app/recipe-details/recipe-details.component.ts:22]: Move this call to `RecipeService`. For example, `recipeService.get(recipeId)`.

File: client/src/app/recipe-details/recipe-details.component.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            this.recipeDetails = result;
+        }, error => console.error(error));
+    }
+}
+
+
+
+
+interface RecipeDetails {
```
Comment (zaidi-01) [inline client/src/app/recipe-details/recipe-details.component.ts:31]: Just `Recipe` is good, it more or less means the same thing but fits better. Also, move this to the `RecipeService` or within that same directory, `client/src/services/recipe`.

File: server/Controllers/RecipeDetailsController.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    private readonly ApplicationDbContext _context;
+    public RecipeDetailsController(ApplicationDbContext context)
+    {
+      _context = context;
+    }
+
+
+// Test function to check if recipe data is being returned or not
+    [HttpGet]
+    public async Task<IActionResult> Test()
```
Comment (zaidi-01) [inline server/Controllers/RecipeDetailsController.cs:24]: All of this should go under the `RecipeController`. "Details" are a subsection of recipes, much like searching, adding and deleting.

File: server/Models/Recipe.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        public required int Id { get; set; }
+        public required string Name { get; set; }
+        public required string Description { get; set; }
+        public required string Ingredients { get; set; }
+        public required string Directions { get; set; }
+        public string? Image { get; set; }
+        public required DateTime Date_Created { get; set; }
+        public required DateTime Date_Updated { get; set; }
+        public string? Categories { get; set; }
+        public string? Cook_Time { get; set; }
```
Comment (zaidi-01) [inline server/Models/Recipe.cs:14]: TODO: figure out how to map these values properly. If nothing works then we'll rename the database columns.

File: client/src/app/recipe-details/recipe-details.component.css [hunk 6]
```diff
@@ -0,0 +1,3 @@
+:host {
```
Comment (zaidi-01) [inline client/src/app/recipe-details/recipe-details.component.css:1]: Make this an `scss` file.

File: client/src/app/recipe-details/recipe-details.component.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+@Component({
+    selector: 'app-recipe-details',
+    templateUrl: `./recipe-details.component.html`,
+    styleUrls: ['./recipe-details.component.css']
+})
+
+
+
+export class RecipeDetailsComponent {
```
Comment (zaidi-01) [inline client/src/app/recipe-details/recipe-details.component.ts:14]: Add the spec file as well.

**Review by zaidi-01 — CHANGES_REQUESTED** [review]

**Review by NilinReza — APPROVED** [review]
No PRs during weekend please! :)

**Review by zaidi-01 — APPROVED** [review]

**Review by JJTomines — APPROVED** [review]

**Pull request merged**